### ISSUE TITLE
about: plt.legend(loc=4) [In[61], page-295, Ch-5]

### ISSUE THREAD

**Issue by gaussEuler** [amueller/introduction_to_ml_with_python#61]
Respected Dr. Muller,
Just for your information: when I use loc=4 in plt.legend, I get the following error:
**UserWarning: Unrecognized location "4". Falling back on "best";** 

By the way, I am able to see the output figure, so the issue is not a problem. However, I just wanted to let you know about this.

matplotlib.__version__
Out: '2.1.0'

Thank you,
Sincerely,
Nikhilesh

**Comment by amueller** [comment]
thanks. Seems like a backward incompatible change. Interesting.

**Comment by amueller** [comment]
Can you please show the complete code and error message? I can't reproduce.

**Comment by gaussEuler** [comment]
Respected Dr. Muller,
Herein is the code from In[61], page 295:
---------------------------------------------------
import matplotlib.pyplot as plt
import mglearn

from mglearn.datasets import make_blobs
X, y = make_blobs(n_samples=(4000, 500), centers=2, cluster_std=[7.0, 2],
                  random_state=22)

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)

from sklearn.svm import SVC
svc = SVC(gamma=0.05).fit(X_train, y_train)

from sklearn.metrics import roc_curve
fpr, tpr, thresholds = roc_curve(y_test, svc.decision_function(X_test))

close_zero = np.argmin(np.abs(thresholds))

plt.plot(fpr[close_zero], tpr[close_zero], 'o', markersize=10,
         label='threshold zero', fillstyle='none', c='k', mew=2)

plt.plot(fpr, tpr, label='ROC curve')
plt.title('ROC curve')
plt.xlabel('FPR')
plt.ylabel('TPR (Recall)')
plt.legend(loc='4')
-------------------------------------------------------------------------------
Herein is the output:
 runfile('C:/Users/Numerical Nikhilesh/.spyder-py3/temp.py', wdir='C:/Users/Numerical Nikhilesh/.spyder-py3')
C:\ProgramData\Anaconda3\lib\site-packages\matplotlib\legend.py:326: UserWarning: Unrecognized location "4". Falling back on "best"; valid locations are
        lower left
        center
        lower right
        upper left
        center right
        right
        center left
        best
        upper center
        upper right
        lower center

  % (loc, '\n\t'.join(self.codes)))

**Note: After the above message, I do get the proper figure as output.**
Changing from loc='4' to loc='best' eliminates the message and only the proper plot (from the book) is shown


Thank you,
Sincerely,
Nikhilesh

**Comment by gaussEuler** [comment]
Please forgive me, I used '4' instead of simply 4

**Comment by gaussEuler** [comment]
It was completely my fault again. Sorry for wasting your time again. I sincerely apologize.
There is no problem with the code

**Comment by amueller** [comment]
np.
